packages/clerk-js/src/ui/components/ApiKeys/ApiKeysTable.tsx

```diff
@@ -0,0 +1,189 @@
+import { useClerk } from '@clerk/shared/react';
+import type { ApiKeyResource } from '@clerk/types';
+import { useEffect, useState } from 'react';
+import useSWR from 'swr';
+
+import { timeAgo } from '../../../utils/date';
+import {
+  Button,
+  descriptors,
+  Flex,
+  Icon,
+  Input,
+  localizationKeys,
+  Spinner,
+  Table,
+  Tbody,
+  Td,
+  Text,
+  Th,
+  Thead,
+  Tr,
+} from '../../customizables';
+import { ThreeDotsMenu } from '../../elements';
+import { useClipboard } from '../../hooks';
+import { Check, Copy, Eye, EyeSlash } from '../../icons';
+
+const useApiKeySecret = ({ apiKeyID, enabled }: { apiKeyID: string; enabled: boolean }) => {
+  const clerk = useClerk();
+
+  return useSWR(enabled ? ['api-key-secret', apiKeyID] : null, () => clerk.getApiKeySecret(apiKeyID));
+};
+
+const CopySecretButton = ({ apiKeyID }: { apiKeyID: string }) => {
+  const [enabled, setEnabled] = useState(false);
+  const { data: apiKeySecret } = useApiKeySecret({ apiKeyID, enabled });
+  const { onCopy, hasCopied } = useClipboard(apiKeySecret ?? '');
+
+  useEffect(() => {
+    if (enabled && apiKeySecret) {
+      onCopy();
+      setEnabled(false);
+    }
+  }, [enabled, apiKeySecret, onCopy]);
+
+  return (
+    <Button
+      variant='ghost'
+      sx={{ margin: 1 }}
+      aria-label={'Copy key'}
+      onClick={() => setEnabled(true)}
+      focusRing={false}
+    >
+      <Icon icon={hasCopied ? Check : Copy} />
+    </Button>
+  );
+};
+
+const SecretInputWithToggle = ({ apiKeyID }: { apiKeyID: string }) => {
+  const [revealed, setRevealed] = useState(false);
+  const { data: apiKeySecret } = useApiKeySecret({ apiKeyID, enabled: revealed });
+
+  return (
+    <Flex
+      center
+      sx={{
+        width: '100%',
+        position: 'relative',
+      }}
+    >
+      <Input
+        type={revealed ? 'text' : 'password'}
+        value={revealed ? (apiKeySecret ?? '') : '•••••••••••••••••••••••••'}
+        readOnly
+        aria-label='API key (hidden)'
+        tabIndex={-1}
+        sx={t => ({
+          paddingRight: t.sizes.$12,
+        })}
+      />
+      <Button
+        variant='ghost'
+        size='sm'
+        sx={{ position: 'absolute', right: 0 }}
+        aria-label={'Show key'}
+        onClick={() => setRevealed(!revealed)}
+        focusRing={false}
```

Comment:
Would still be helpful to have a visible focus ring, maybe can be added via focusVisible if you don't want to see it on click?

Reply:
I think this would work:

```jsx
'&:focus-visible': {
    ...common.focusRingStyles(t),
},
```